### PR TITLE
Added Linear Search Issue#3435

### LINKED ISSUE #3435: Linear Search Algorithm in Kotlin
## 💥 Proposal

(A clear and concise description of what the proposal is.)
I would like to implement the linear search algorithm in Kotlin .
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

(Write your answer here.)
yes

please assign this issue to me under GSSoC'21

### PR DESCRIPTION
**Pull request by nagasrihith608**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
yes
### Description
Added Linear search algo in kotlin.
(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Fixes #3435 
(Write your answer here.)

### CODE REVIEW

File: Kotlin/search/LinearSearch.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
+//This is a program that allows us to do linear search in kotlin
```
Comment (ankitaggarwal23) [inline Kotlin/search/LinearSearch.kt]: Add few more description

File: Kotlin/search/LinearSearch.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+import java.util.*
+fun main(){
+
+   val scan = Scanner(System.`in`)
+   
+   print("Enter the number of elements :")
+   //Here we take the size of the input array/List
```
Comment (ankitaggarwal23) [inline Kotlin/search/LinearSearch.kt:31]: Remove the redundant comments

File: Kotlin/search/LinearSearch.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+   val array = IntArray(scan.nextInt())
+   
+   print("Enter the elements :")
+   //Taking the elements and storing them in array
```
Comment (ankitaggarwal23) [inline Kotlin/search/LinearSearch.kt:35]: Remove the redundant comments

File: Kotlin/search/LinearSearch.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+   for(element in array.indices){
+     array[element] = scan.nextInt()
+   } 
+    
+   print("Enter the number you want to search for :")
+   //The number we want to search for
```
Comment (ankitaggarwal23) [inline Kotlin/search/LinearSearch.kt:41]: Remove the redundant comments

File: Kotlin/search/LinearSearch.kt [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+   }
+   //Print this message if the element is not found in the array/List
+   if(i==0){
+      print("The number given is not there in the given List")
+   }
+
+}
+
+/*
+
```
Comment (ankitaggarwal23) [inline Kotlin/search/LinearSearch.kt]: Remove the space

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
- Add Time and Space Complexity at the end of the Code

**Comment by nagasrihith608** [comment]
hey, @ankitaggarwal23 I had updated the requested changes please have a look at them.

**Comment by nagasrihith608** [comment]
@HarshCasper please merge this. Thank you.

File: Kotlin/search/LinearSearch.kt [hunk 6]
```diff
@@ -0,0 +1,59 @@
+/* Linear search algorithm is an algorithm which checks if an element is present in the given list by comparing it with every element in the list.
+   Linear search is the simplest search algorithm; it is a special case of brute-force search and in worst-case, cost is proportional to the number of elements in the list. Hence name is Linear search.
+   This is a linear search algorithm in Kotlin.
+*/
+
+import java.util.*
+fun main(){
```
Comment (HarshCasper) [inline Kotlin/search/LinearSearch.kt]: Don't put the main logic in the main() function

File: Kotlin/search/LinearSearch.kt [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  we have 2 at index : 1
+  we have 2 at index : 2
+
+  Enter the number of elements :4
+  Enter the elements :1 2 3 4
+  Enter the number you want to search for :5
+  The number given is not there in the given List 
+
+*/
+
```
Comment (HarshCasper) [inline Kotlin/search/LinearSearch.kt:57]: Remove any extra new lines

File: Kotlin/search/LinearSearch.kt [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  Enter the number of elements :4
+  Enter the elements :1 2 2 4
+  Enter the number you want to search for :2
+  we have 2 at index : 1
+  we have 2 at index : 2
+
+  Enter the number of elements :4
+  Enter the elements :1 2 3 4
+  Enter the number you want to search for :5
+  The number given is not there in the given List
```
Comment (HarshCasper) [inline Kotlin/search/LinearSearch.kt]: Use only one sample I/O.

File: Kotlin/search/LinearSearch.kt [hunk 9]
```diff
@@ -0,0 +1,59 @@
+/* Linear search algorithm is an algorithm which checks if an element is present in the given list by comparing it with every element in the list.
+   Linear search is the simplest search algorithm; it is a special case of brute-force search and in worst-case, cost is proportional to the number of elements in the list. Hence name is Linear search.
+   This is a linear search algorithm in Kotlin.
+*/
+
```
Comment (HarshCasper) [inline Kotlin/search/LinearSearch.kt]: Make it more readable

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Pull request closed without merging**